client_apis/python/src/tests/integration/alerts_test.py

```diff
@@ -0,0 +1,212 @@
+#
+# CARBON BLACK API TESTS - alerts
+# Copyright, Bit9, Inc 2015
+#
+
+""" These tests require CarbonBlack Enterprise Server to be installed.
+    You can run this script by passing the server URL and user API token
+    as parameters.
+"""
+
+import unittest
+import sys
+import os
+import json
+import requests
+import time
+import paramiko, base64
+
+
+if __name__ == '__main__':
+    sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(os.path.abspath(__file__)), "../../")))
+
+from cbapi.cbapi import CbApi
+from datetime import datetime
+
+cb = None
+
+class ShhHelper:
+    @classmethod
+    def execute_command(cls, server, user, password, command):
+        client = paramiko.SSHClient()
+        client.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+        client.connect(server, username=user, password=password)
+
+        transport = client.get_transport()
+        session = transport.open_session()
+        session.set_combine_stderr(True)
+        session.get_pty()
+
+        session.exec_command("sudo -k %s" % command)
+        stdin = session.makefile('wb', -1)
+        stdout = session.makefile('rb', -1)
+        stdin.write(password +'\n')
+        stdin.flush()
+
+        for line in stdout.read().splitlines():
+            print '... ' + line.strip('\n')
+
+        client.close()
+
+class CbApiAlertTestCase(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.watchlist_name = "alerts_watchlist_%s" % datetime.now().strftime('%m%d%Y%H%M%S')
+
+        watchlist = cb.watchlist_add(
+            type="events",
+            name=cls.watchlist_name,
+            search_query="b.urlver=1&q=process_name%3Achrome.exe&sort=&rows=10&start=0",
+            readonly=False
+        )
+
+        cls.watchlist_id = watchlist["id"]
+
+        cb.watchlist_action_add(
+            watchlist_id=cls.watchlist_id,
+            action_type_id=3,
+            email_recipient_user_ids=[1]
+        )
+
+        ShhHelper.execute_command(server, user, password, '/usr/bin/python -m cb.maintenance.job_runner --master -s watchlist_search')
+        time.sleep(20)
+        ShhHelper.execute_command(server, user, password, '/usr/share/cb/cbsolr --soft-commit')
+        time.sleep(5)
+
+    def test_all_positive_cases(self):
```

Comment:
Break up your test cases into separate methods so that if this gets run through unit test, it's clear what method(s) fail